src/main/java/org/alliancegenome/curation_api/jobs/executors/GeneFmsExecutor.java

```diff
@@ -1,66 +1,103 @@
 package org.alliancegenome.curation_api.jobs.executors;
 
 import java.io.FileInputStream;
+import java.util.*;
+import java.util.stream.Collectors;
 import java.util.zip.GZIPInputStream;
 
 import javax.enterprise.context.ApplicationScoped;
 import javax.inject.Inject;
 
+import org.alliancegenome.curation_api.dao.GeneDAO;
 import org.alliancegenome.curation_api.exceptions.ObjectUpdateException;
 import org.alliancegenome.curation_api.exceptions.ObjectUpdateException.ObjectUpdateExceptionData;
+import org.alliancegenome.curation_api.model.entities.Gene;
 import org.alliancegenome.curation_api.model.entities.bulkloads.*;
 import org.alliancegenome.curation_api.model.ingest.fms.dto.*;
 import org.alliancegenome.curation_api.response.*;
 import org.alliancegenome.curation_api.services.GeneService;
 import org.alliancegenome.curation_api.util.ProcessDisplayHelper;
+import org.apache.commons.collections4.ListUtils;
 
 import lombok.extern.jbosslog.JBossLog;
 
 @JBossLog
 @ApplicationScoped
 public class GeneFmsExecutor extends LoadFileExecutor {
 
+    @Inject GeneDAO geneDAO;
```

Comment:
So typically its better for the service (geneService) to know how to call the DAO vs using the DAO directly. The service should have a method like `getGeneByCurie(String curie)` which then calls the DAO findByField that way the logic is in the service and can be reused.

Reply:
Seems to me like you wrote [the removal code for FMS Disease annotation data](https://github.com/alliance-genome/agr_curation/commit/2b8f805471e6c5231958691fc1a605b3dceb3a9d) the same way as I did here, which I essentially just replicated to Allele- (previously), Gene- and AGM removal (this PR)?

Reply:
This code was not "created" in that PR it was moved. Do a search for: `runLoad(String taxonId, List<GeneDiseaseAnnotationDTO> annotations)` and you will see it was moved from the Service to the Executer. Which at that time more refactor should have happened... My point with this, is that for new code we need to keep these kind of principles in mind.